pkg/controllerutils/worker.go

```diff
@@ -20,38 +20,48 @@ import (
 	"sync"
 	"time"
 
+	"github.com/go-logr/logr"
 	"k8s.io/apimachinery/pkg/types"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/tools/cache"
 	"k8s.io/client-go/util/workqueue"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
-	"sigs.k8s.io/controller-runtime/pkg/runtime/inject"
 
 	"github.com/gardener/gardener/pkg/logger"
 )
 
-// CreateWorker creates and runs a worker thread that just processes items in the
-// specified queue. The worker will run until stopCh is closed. The worker will be
+// WorkerOptions are options for a controller's worker.
+type WorkerOptions struct {
+	logger logr.Logger
+}
+
+// WorkerOption is a func that mutates WorkerOptions.
+type WorkerOption func(*WorkerOptions)
+
+// WithLogger configures the logr.Logger to use for a controller worker.
+// If set, a logger preconfigured with the name and namespace field will be injected into the context.Context on
+// each Reconcile call. It can be retrieved via log.FromContext.
+func WithLogger(logger logr.Logger) WorkerOption {
+	return func(options *WorkerOptions) {
+		options.logger = logger
+	}
+}
+
+// CreateWorker creates and runs a worker goroutine that just processes items in the
+// specified queue. The worker will run until ctx is cancelled. The worker will be
 // added to the wait group when started and marked done when finished.
-// The given context is injected into the `reconciler` if it implements `inject.Stoppable`.
-// Optionally passed inject functions are called with the `reconciler` but potentially returned errors are disregarded.
-func CreateWorker(ctx context.Context, queue workqueue.RateLimitingInterface, resourceType string, reconciler reconcile.Reconciler, waitGroup *sync.WaitGroup, workerCh chan<- int, injectFn ...inject.Func) {
-	fns := append(injectFn, func(i interface{}) error {
-		_, err := inject.StopChannelInto(ctx.Done(), i)
-		return err
-	})
-
-	for _, f := range fns {
-		if err := f(reconciler); err != nil {
-			logger.Logger.Errorf("An error occurred while reconciler injection: %v", err)
-		}
```

Comment:
Why has this code been removed? Is there a controller-runtime change that makes it obsolete (or just not a good idea)?
Do I understand correctly that this change is not really related to logging? If yes, is it a good idea to have it in this PR?

Reply:
I removed it because this code was unused and it was easier to do my changes (optionally injecting a logger) by removing this first.